### PR TITLE
IRV update

### PR DESCRIPTION
**Pull request by miaecle**
This PR adds in:

- Tensorgraph version of IRV with identical structure

- TensorflowLogisticRegression is removed and standard sklearn logistic regression is used in examples and benchmark

### CODE REVIEW

File: deepchem/models/tensorflow_models/IRV.py [hunk 1]
```diff
@@ -2,32 +2,142 @@
...
+    self.IRVLayer = IRVLayer
+    self.penalty = penalty
+    super(IRVRegularize, self).__init__(**kwargs)
+
+  def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
+    assert self.IRVLayer.out_tensor is not None, "IRVLayer must be built first"
+    out_tensor = tf.nn.l2_loss(self.IRVLayer.W) + \
+        tf.nn.l2_loss(self.IRVLayer.V) + tf.nn.l2_loss(self.IRVLayer.b) + \
+        tf.nn.l2_loss(self.IRVLayer.b2)
+    out_tensor = out_tensor * self.penalty
```
Comment (miaecle) [inline deepchem/models/tensorflow_models/IRV.py]: IRV has l2 regularization, is there a better way to handle these extra loss values?

Comment (rbharath) [reply deepchem/models/tensorflow_models/IRV.py]: I think this looks fine.

File: deepchem/models/tensorgraph/layers.py [hunk 2]
```diff
@@ -1184,6 +1184,26 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
     return out_tensor
 
 
+class Sigmoid(Layer):
```
Comment (rbharath) [inline deepchem/models/tensorgraph/layers.py]: Could you add a docstring for this class along with a usage example?

File: deepchem/models/tensorflow_models/IRV.py [hunk 3]
```diff
@@ -2,32 +2,142 @@
...
+    if set_tensors:
+      self.out_tensor = out_tensor
+    return out_tensor
+
+
+class Slice(Layer):
+  """ Choose a slice of input given axis and order
```
Comment (rbharath) [inline deepchem/models/tensorflow_models/IRV.py]: Could you add a usage example for this class in the docstring?

File: deepchem/models/tensorgraph/layers.py [hunk 4]
```diff
@@ -1531,6 +1551,27 @@ def create_tensor(self, in_layers=None, set_tensors=True, **kwargs):
     return out_tensor
 
 
+class SigmoidCrossEntropy(Layer):
```
Comment (rbharath) [inline deepchem/models/tensorgraph/layers.py]: Could you add a usage example and docstring for this class?

File: deepchem/models/tensorflow_models/IRV.py [hunk 5]
```diff
@@ -2,32 +2,142 @@
...
-from deepchem.models.tensorflow_models.lr import TensorflowLogisticRegression
+from deepchem.models.tensorgraph.tensor_graph import TensorGraph
+from deepchem.models.tensorgraph.layers import Layer, SigmoidCrossEntropy, \
+    Sigmoid, Feature, Label, Weights, Concat, WeightedError
+from deepchem.models.tensorgraph.layers import convert_to_layers
+from deepchem.trans import undo_transforms
 
 
-class TensorflowMultiTaskIRVClassifier(TensorflowLogisticRegression):
+class IRVLayer(Layer):
```
Comment (rbharath) [inline deepchem/models/tensorflow_models/IRV.py]: Could you add a unit test for these and the other layers added?

**Comment by rbharath** [comment]
@miaecle Thanks for porting! Made a few comments with requests for more docs and tests.

**Comment by miaecle** [comment]
@rbharath Thanks for the review! Just added some more docstrings and unit tests, let me know if anything else could be improved.

**Comment by rbharath** [comment]
@miaecle Thanks for the fixes! Would you mind adding tests that the new layers can be pickled? Sorry, I should have mentioned this on the original review!

As a heads up, I'm going to merge in #992 since that one's been open for a while. There might be some minor merge conflicts heading your way.

**Comment by miaecle** [comment]
@rbharath  No problem, I will handle it.

**Comment by rbharath** [comment]
Thanks for the fixes!

LGTM
